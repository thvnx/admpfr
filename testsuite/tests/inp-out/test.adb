with Ada.Text_IO; use Ada.Text_IO;
with Admpfr;      use Admpfr;

procedure Test is
   F, I  : File_Type;
   M, N : Mpfloat;
begin
   Create (F, Out_File, "tt.txt");
   Create (I, Out_File, "fpif.txt");
   M.Const_Catalan;
   Out_Str (F, M);
   Fpif_Export (I, M);
   Put_Line (F, "");
   M.Const_Euler;
   Out_Str (F, M);
   Fpif_Export (I, M);
   Put_Line (F, "");
   M.Const_Log2;
   Out_Str (F, M);
   Fpif_Export (I, M);
   Put_Line (F, "");
   Close (F);
   Close (I);

   Open (F, In_File, "tt.txt");
   Open (I, In_File, "fpif.txt");
   Inp_Str (N, F);
   Put_Line (N'Image);
   Fpif_Import (N, I);
   Put_Line (N'Image);
   N.Dump;
   Inp_Str (N, F);
   Put_Line (N'Image);
   Fpif_Import (N, I);
   Put_Line (N'Image);
   N.Dump;
   Inp_Str (N, F);
   Put_Line (N'Image);
   Fpif_Import (N, I);
   Put_Line (N'Image);
   N.Dump;
   Close (F);

   Out_Str (Standard_Output, N);

end Test;
