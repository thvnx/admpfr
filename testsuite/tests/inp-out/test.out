9.1596559417721901e-01
9.1596559417721901e-01
0.11101010011111001011100010011111010000001001101011101E0
5.7721566490153287e-01
5.7721566490153287e-01
0.10010011110001000110011111100011011111011011000011001E0
6.9314718055994529e-01
6.9314718055994529e-01
0.10110001011100100001011111110111110100011100111101111E0
6.9314718055994529e-1